CONTRIBUTING.md

```diff
@@ -0,0 +1,118 @@
+  should be carefully reviewed and integrated. Ideally, several developers would
+  test that the changes don't break extensions.
+
+#### How to integrate a PR ?
+
+Getting your contributions integrated is relatively straightforward, here
+is the checklist:
+
+* All tests pass
+* Consensus is reached. This usually means that at least two reviewers approved
+the changes (or added a `LGTM` comment) and at least one business day passed
+without anyone objecting. `LGTM` is an acronym for _Looks Good to Me_.
+
+Next, there are two scenarios:
+* You do NOT have push access: A Slicer core developer will integrate your PR. If
+  you would like to speed up the integration, do not hesitate to send a note on
+  the mailing list.
+* You have push access: Follow [Integrating topic from external contributor](https://www.slicer.org/wiki/Slicer:git-svn#Integrating_topic_from_external_contributor)
+  instructions on the wiki.
+
+
+#### Automatic testing of pull requests
+
+Every pull request is tested automatically using CircleCI each time you push a
+commit to it. The Github UI will restrict users from merging pull requests until
+the CI build has returned with a successful result indicating that all tests have
+passed.
+
+The testing infrastructure is described in details in the
+[3D Slicer Improves Testing for Pull Requests Using Docker and CircleCI](https://blog.kitware.com/3d-slicer-improves-testing-for-pull-requests-using-docker-and-circleci/)
+blog post.
+
+
+#### Decision-making process
+
+1. Given the topic of interest, initiate discussion on the [mailing list](http://massmail.spl.harvard.edu/mailman/listinfo/slicer-devel).
+
+2. Identify a small circle of community members that are interested to study the
+   topic in more depth.
+
+3. Take the discussion off the general list, work on the analysis of options and
+   alternatives, summarize findings on the wiki or similar. [Labs](https://www.slicer.org/wiki/Documentation/Labs)
+   page are usually a good ground for such summary.
+
+4. Announce on the mailing list the in-depth discussion of the topic for the
+   [Slicer Community hangout](https://www.slicer.org/wiki/Documentation/Nightly/Developers/Meetings),
+   encourage anyone that is interested in weighing in on the topic to join the
+   discussion. If there is someone who is interested to participate in the discussion,
+   but cannot join the meeting due to conflict, they should notify the leaders of
+   the given project and identify the time suitable for everyone.
+
+5. Hopefully, reach consensus at the hangout and proceed with the agreed plan.
+
+
+*The initial version of these guidelines was established during the [winter
+ project week 2017](http://www.na-mic.org/Wiki/index.php/2017_Winter_Project_Week/UpdatingCommunityForums).*
+
+#### Benevolent dictators for life
+
+The [benevolent dictators](https://en.wikipedia.org/wiki/Benevolent_dictator_for_life) can bypass the regular
+process and integrate changes to keep the platform healthy.
```

Comment:
, integrate changes to keep the platform healthy and help interpret or address conflict related to the contribution guidelines.